src/hooks/useFetch/useFetch.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setTimeout(() => {
```

Comment:
Unsure about the usage of setTimeout here. Can you elaborate it please?
Moreover, please prefer using axios over vanilla fetch. Error handling in axios is much more simple.